linera-core/src/client/mod.rs

```diff
@@ -1207,6 +1208,7 @@ where
         &self,
         certificate: Certificate,
         mode: ReceiveCertificateMode,
+        nodes: Option<Vec<RemoteNode<P::Node>>>,
```

Comment:
> Download the blob in question from the validator that sent us the certificate and is therefore expected to have it.

This code looks like it does the opposite, i.e. downloading the certificate from the validator that has the blob. What am I missing?

Reply:
The certificate itself—the one we downloaded _because_ of the blob in question—is passed into this function. But the function also automatically downloads all the ancestors and blobs, if needed.

It currently downloads those from _all_ validators, sequentially. With this change, if a node is provided explicitly, it downloads them only from that node.

Reply:
Yes, exactly, that's a more general and far-reaching discussion we should have at some point: The right tradeoff between latency (ask all validators everything at the same time, take the first response) and bandwidth (ask them in sequence).

But at least it triggers exactly that bottleneck way less frequently that before: Before, we were simply asking the _wrong_ validator often, and then, due to #3033, waiting a long time for it.

Reply:
re: tradeoff: We don't need to be faster than the fastest validator.